Brokerages/InteractiveBrokers/InteractiveBrokersSymbolMapper.cs

```diff
@@ -206,5 +209,55 @@ private string GetMappedTicker(Symbol symbol)
 
             return ticker;
         }
+
+        /// <summary>
+        /// Parses a contract for options with malformed data.
+        /// Malformed data usually manifests itself by having "0" assigned to some values
+        /// we expect, like the contract's expiry date. The contract is returned by IB
+        /// like this, usually due to a high amount of data subscriptions that are active
+        /// in an account, surpassing IB's imposed limit. Read more about this here: https://interactivebrokers.github.io/tws-api/rtd_fqa_errors.html#rtd_common_errors_maxmktdata
+        ///
+        /// We are provided a string in the Symbol in malformed contracts that can be
+        /// parsed to construct the clean contract, which is done by this method.
+        /// </summary>
+        /// <param name="malformedContract">Malformed contract (for options), i.e. a contract with invalid values ("0") in some of its fields</param>
+        /// <returns>Clean Contract for the option</returns>
+        /// <remarks>
+        /// The malformed contract returns data similar to the following when calling <see cref="InteractiveBrokersBrokerage.GetContractDetails"/>:
+        /// OPT SPY JUN2021 350 P [SPY 210618P00350000 100] USD 0 0 0
+        ///
+        /// ... which the contents inside [] follow the pattern:
+        ///
+        /// [SYMBOL YY_MM_DD_OPTIONRIGHT_STRIKE(divide by 1000) MULTIPLIER]
+        /// </remarks>
+        public static Contract ParseMalformedContractOptionSymbol(Contract malformedContract)
+        {
+            var contractInfoSplit = malformedContract.Symbol.Substring(malformedContract.Symbol.IndexOf('['))
+                .Replace("[", "")
+                .Replace("]", "")
+                .Split(' ');
+
+            var contractSymbol = contractInfoSplit[0];
+            var contractSpecification = contractInfoSplit[1];
+            var multiplier = contractInfoSplit[2];
+            var year = "20" + contractSpecification.Substring(0, 2);
+            var month = contractSpecification.Substring(2, 2);
+            var day = contractSpecification.Substring(4, 2);
+            var contractRight = contractSpecification[6] == 'C' ? IB.RightType.Call : IB.RightType.Put;
+            var contractStrike = long.Parse(contractSpecification.Substring(7), CultureInfo.InvariantCulture) / 1000.0;
+
+            return new Contract
+            {
+                Symbol = contractSymbol,
+                Multiplier = multiplier,
+                LastTradeDateOrContractMonth = year + month + day,
+                Right = contractRight,
+                Strike = contractStrike,
+                Exchange = "SMART",
+                SecType = IB.SecurityType.Option,
+                IncludeExpired = false,
+                Currency = "USD"
```

Comment:
Nice!
From `OPT SPY JUN2021 350 P [SPY 210618P00350000 100] USD 0 0 0` believe `Currency` is being set by IB suggest we use the provided value, same for security type. This will allow us to reuse the same logic in the case of a FOP which I think we should cover too calling this same method?